diff --git a/CMakeLists.txt b/CMakeLists.txt
index f2640e2..048d0f5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,7 @@ PROJECT(matrix_multiply_example)
 
 SET( CMAKE_BUILD_TYPE RELEASE )
 
-SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++1y" )
+SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++1y -fopenmp" )
 SET( CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -pedantic -DGLIBCXX_DEBUG" )
 SET( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}" )
 
diff --git a/main.cpp b/main.cpp
index 13fe4fc..d623f2a 100644
--- a/main.cpp
+++ b/main.cpp
@@ -28,6 +28,7 @@ int main(int argc, char** argv){
   int iterations = max_iterations;
 
   while( iterations-- ) {
+#pragma omp parallel for
     for ( int y = 0; y < Y; y++ ){
       for ( int x = 0; x < X; x++ ) { 
         targetVector[y] += matrix[x+y*X] * inputVector[x];
